### PR TITLE
LibRetro.RetroArch.Nightly

### PR DESCRIPTION
**Pull request by fpscan**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`? 
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.1 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.1.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

I've been trying to bring to bring RetroArch to windows package manager but I need some help at this moment. So we have removed to directx9.0 pop-up before installation but it gives UAC now. I made this PR with YAML script and this is how we wanted to seen. Is there any way around to get rid of UAC either with WinGet or something else?

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/59175)

### CODE REVIEW

File: manifests/l/Libretro/RetroArch/Nightly/1.10.3/Libretro.RetroArch.Nightly.locale.en-US.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Created with YamlCreate.ps1 v2.1.2 $debug=NVS1.5-1-22598-1
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.1.0.schema.json
+
+PackageIdentifier: Libretro.RetroArch.Nightly
+PackageVersion: 1.10.3
+PackageLocale: en-US
+Publisher: LibRetro
```
Comment (ItzLevvie) [inline manifests/l/Libretro/RetroArch/Nightly/1.10.3/Libretro.RetroArch.Nightly.locale.en-US.yaml:7]:
```suggestion
Publisher: Libretro
```
This must match Control Panel.

File: manifests/l/Libretro/RetroArch/Nightly/1.10.3/Libretro.RetroArch.Nightly.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Created with YamlCreate.ps1 v2.1.2 $debug=NVS1.5-1-22598-1
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.version.1.1.0.schema.json
+
+PackageIdentifier: Libretro.RetroArch.Nightly
+PackageVersion: 1.10.3
```
Comment (ItzLevvie) [inline manifests/l/Libretro/RetroArch/Nightly/1.10.3/Libretro.RetroArch.Nightly.yaml:5]:
```suggestion
PackageVersion: 1.10.3.0
```
This must match Control Panel.

File: manifests/l/Libretro/RetroArch/Nightly/1.10.3/Libretro.RetroArch.Nightly.locale.en-US.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# Created with YamlCreate.ps1 v2.1.2 $debug=NVS1.5-1-22598-1
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.1.0.schema.json
+
+PackageIdentifier: Libretro.RetroArch.Nightly
+PackageVersion: 1.10.3
```
Comment (ItzLevvie) [inline manifests/l/Libretro/RetroArch/Nightly/1.10.3/Libretro.RetroArch.Nightly.locale.en-US.yaml:5]:
```suggestion
PackageVersion: 1.10.3.0
```
This must match Control Panel.

File: manifests/l/Libretro/RetroArch/Nightly/1.10.3/Libretro.RetroArch.Nightly.installer.yaml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# Created with YamlCreate.ps1 v2.1.2 $debug=NVS1.5-1-22598-1
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.1.0.schema.json
+
+PackageIdentifier: Libretro.RetroArch.Nightly
+PackageVersion: 1.10.3
```
Comment (ItzLevvie) [inline manifests/l/Libretro/RetroArch/Nightly/1.10.3/Libretro.RetroArch.Nightly.installer.yaml:5]:
```suggestion
PackageVersion: 1.10.3.0
```
This must match Control Panel.

**Review by ItzLevvie — CHANGES_REQUESTED** [review]

**Comment by ItzLevvie** [comment]
> So we have removed to directx9.0 pop-up before installation but it gives UAC now.

Azure Pipelines is running in administrator mode so the installer isn't running into any UAC issues.

`Validation-SmartScreen` is also no longer being used as SmartScreen issues delayed developers from adding new versions into the package repository for a few days or even a few weeks so that was killed off.

It only has issues downloading the installer.

Delivery Optimization (default):
```
Enter a pull request number: 59175

Manifest validation succeeded.
Found RetroArch Nightly [Libretro.RetroArch.Nightly] Version 1.10.3
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Downloading https://buildbot.libretro.com/nightly/windows/x86_64/RetroArch-Win64-setup.exe
  ██████████████████████████████   173 MB /  173 MB
Installer hash does not match; this cannot be overridden when running as admin

Press any key to continue . . .
```

WinINet
```
Enter a pull request number: 59175

Manifest validation succeeded.
Found RetroArch Nightly [Libretro.RetroArch.Nightly] Version 1.10.3
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Downloading https://buildbot.libretro.com/nightly/windows/x86_64/RetroArch-Win64-setup.exe
  ██████████████████████████████   173 MB /  173 MB
Successfully verified installer hash
Starting package install...
Successfully installed

Name          : RetroArch
Publisher     : Libretro
Version       : 1.10.3.0
ProductCode   : RetroArch
Registry Path : Computer\HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall\RetroArch
Uninstall     : winget uninstall --id "RetroArch"

Press any key to continue . . .
```

**Comment by ItzLevvie** [comment]
I'm no longer seeing any issues now with Delivery Optimization but it may still happen in Azure Pipelines.

```
Enter a pull request number: 59175

Manifest validation succeeded.
Found RetroArch Nightly [Libretro.RetroArch.Nightly] Version 1.10.3
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Downloading https://buildbot.libretro.com/nightly/windows/x86_64/RetroArch-Win64-setup.exe
  ██████████████████████████████   173 MB /  173 MB
Successfully verified installer hash
Starting package install...
Successfully installed

Name          : RetroArch
Publisher     : Libretro
Version       : 1.10.3.0
ProductCode   : RetroArch
Registry Path : Computer\HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall\RetroArch
Uninstall     : winget uninstall --id "RetroArch"

Press any key to continue . . .
```

`settings.json`:
```
{
    "network": {"downloader": "do"},
    "source": {"autoUpdateIntervalInMinutes": 0},
    "visual": {"progressBar": "rainbow"}
}
```

File: manifests/l/Libretro/RetroArch/Nightly/1.10.3/Libretro.RetroArch.Nightly.yaml (hunk 2 above)
Comment (fpscan) [reply manifests/l/Libretro/RetroArch/Nightly/1.10.3/Libretro.RetroArch.Nightly.yaml:5]: appropriate version is `1.10.3`

**Comment by fpscan** [comment]
@ItzLevvie Hi, thank you for your review. I'm not sure what to do at this moment. I was thinking UAC was the problem but seems like it is not. How can I solve `Sha256 Hash does not match the installer.` since this is new it didn't happened for my old PRs that I closed for this one. Can this be manually approved?

File: manifests/l/Libretro/RetroArch/Nightly/1.10.3/Libretro.RetroArch.Nightly.yaml (hunk 2 above)
Comment (ItzLevvie) [reply manifests/l/Libretro/RetroArch/Nightly/1.10.3/Libretro.RetroArch.Nightly.yaml:5]: Would you be able to modify your installer to reflect this?

**Review by ItzLevvie — CHANGES_REQUESTED** [review]
https://github.com/microsoft/winget-pkgs/pull/59175#discussion_r862382108

I'm still seeing `1.10.3.0` in Control Panel.

"The current best practice is to use the value reported in Add / Remove Programs (Control Panel) when this version of the package is installed. In some cases, packages do not report a version resulting in an upgrade loop or other unwanted behavior." - https://github.com/microsoft/winget-pkgs/blob/master/doc/manifest/schema/1.1.0/version.md

File: manifests/l/Libretro/RetroArch/Nightly/1.10.3/Libretro.RetroArch.Nightly.yaml (hunk 2 above)
Comment (fpscan) [reply manifests/l/Libretro/RetroArch/Nightly/1.10.3/Libretro.RetroArch.Nightly.yaml]: Instead I changed the packageversion to `1.10.3.0`

**Comment by fpscan** [comment]
> Hello @fpscan, The package manager bot determined that the metadata was not compliant.
> 
> Please verify the manifest file is compliant with the package manager [1.0 manifest specification](https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv1.0.md). Make sure the ID is of the form publisher.appname and that the folder structure is manifests\partition\publisher\appname\version. Note: The path and "PackageIdentifier" are case sensitive. Be sure to use a tool like VSCode (https://code.visualstudio.com/) to make sure the manifest YAML syntax is correct.
> 
> You could also try our [Windows Package Manager Manifest Creator Preview](https://github.com/microsoft/winget-create).
> 
> For details on the specific error, see the details link below in the build pipeline.

@ItzLevvie Should I replace folder name to `1.10.3.0` from `1.10.3`?

**Comment by ItzLevvie** [comment]
> Should I replace folder name to 1.10.3.0 from 1.10.3?

Yes.

**Comment by fpscan** [comment]
I'm sorry but didn't changed the result. Is there anyway around to fix the
hash check?

1 May 2022 Pzr 09:09 tarihinde azure-pipelines[bot] <
***@***.***> şunu yazdı:

**Comment by fpscan** [comment]
Hi @ItzLevvie , maybe this can be start point for the fix.  

Here is our hash;

image

Here is Azure's hash;

image

**Review by ItzLevvie — APPROVED** [review]

**Pull request closed without merging**